From 995d36815240fe6e45c5c1e848173f3ad328b960 Mon Sep 17 00:00:00 2001
From: Dorian Krause <github@91df63fb.com>
Date: Fri, 12 Jun 2020 21:54:13 +0200
Subject: [PATCH 2/5] Bug fix: Fix typo in function name

---
 watchman/clingy_file.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/watchman/clingy_file.cxx b/watchman/clingy_file.cxx
index 074a1b3..c972bc8 100644
--- a/watchman/clingy_file.cxx
+++ b/watchman/clingy_file.cxx
@@ -203,7 +203,7 @@ bool _clingy_file_attach(const char *arg, char *mountpoint, char *fstype, char *
 	return 0;
 }
 
-bool _cling_file_filesystem_is_mounted(const char *mountpoint, const char *fstype, const char *source)
+bool _clingy_file_filesystem_is_mounted(const char *mountpoint, const char *fstype, const char *source)
 {
 	int err;
 	int tmp;
-- 
2.25.4

